### DISCUSSION TITLE
Root dir selection in projects with multiple LSP workspaces?

### DISCUSSION THREAD

**Discussion by phinze** [LazyVim/LazyVim#143] [Q&A] [answered]
First of all, thanks for the project! I've been enjoying playing with it for the past few days. 🎉 

I've been working in a project that uses a bunch of ember engines in various subdirectories, which each has its own LSP root. My problem is that once I have a file open from one of these engines (which is most of the time), it seems like root dir and cwd operations both behave the same. It's normal for me to need to hop between two or three engines when working, and I keep getting stuck in one.

Is there a way to "force" the root dir for a project so my find files / neotree / etc operations always execute from the git project root and not the nearest LSP root?

**Comment by phinze** [comment]
(It's also possible I'm just misunderstanding the semantics of root dir vs cwd, and I should just be using CWD operations for a situation like this. If so let me know!)

**Comment by folke** [comment]
What functionality is stuck? Is it neo-tree? For neo-tree I did fix something since it always changed the `cwd`. That should now no longer happen.

**Comment by folke** [comment, marked as answer]
I just created a new release that contains those neo-tree fixes, so that probably fixeds your issue

**Comment by phinze** [comment]
Thanks @folke that fixed it! Using the `cwd` versions of operations from this project now works as expected.

**Comment by etiennelacoursiere** [comment]
I'm having the exact same issue i think and i'm pretty sure i'm up to date.

I have a repo which has two project in it `/api` and `/app` 

The API is in Elixir and the App is in Typescript. 

When i open the project from the root and do `space space` i have the Git Files telescope picker and can search for files in both projects. From here, if i open a `.tsx` file, then when i do `space space` again it now open the Find Files Telescope picker and it doesn't see my `.ex` files from the API.

Im guessing some plugin is messing with the root of my project.

EDIT: Oh i just realized there is a `space f F`. that does what i want.

**Reply by folke** [reply]
Yep, indeed. That's but design

**Comment by phinze** [comment]
Hi @etiennelacoursiere - I think you might actually be seeing the intended behavior.

`<space><space>` is mapped to "Find files (root dir)" which is meant to search based on the current LSP's root. If you want to find files across multiple LSP roots, you want `<space>fF` which is mapped to "Find files (cwd)" and should search everything from the base dir of the project.

The same CWD vs LSP Root Dir split exists for neo-tree as well. That's the difference between `<space>E` and `<space>e`.

There was previously a problem where the CWD was getting changed by neo-tree so even `<space>fF` was getting confused, but that's now fixed.

Hope this helps!

EDIT: Ha you figured it out while I was typing! Nice 😄
